### DISCUSSION TITLE
cache 機制

### DISCUSSION THREAD

**Discussion by Rubyrubylai** [Lidemy/forum#113] [其他問題]
資料: https://blog.techbridge.cc/2017/06/17/cache-introduction/#post-comment-wrapper
您好，拜讀完您寫的文章以及參考其相關的文章後，嘗試做 client 及 server 之間的 cache，實作後，卻發現沒辦法讓網站 cache 住，想請教是否有設定錯誤，或著是我的理解有誤，謝謝。
環境架構: node.js + Express框架 + react(前後端分離）
實作方式:
在後端 API 設定
```
router.post('/listProductByHash', async(req, res) => {
  res.setHeader("Cache-Control", "public, max-age = 86400");
}
```
結果:
從 chrome dev tool 上 status code 只有回傳 200，沒有 from memory cache

![image](https://user-images.githubusercontent.com/44698002/118439975-8f3ae880-b719-11eb-9b65-ef4be5c4d6f0.png)

**Comment by cwc329** [comment]
我不知道問題的解法，
不過有點好奇你為何要 cache post method 的 response？

**Reply by Rubyrubylai** [reply]
因為資料過長無法使用 GET 來傳遞，但實際要做的動作其實也只是要渲染畫面
不過這也確實是我沒發現的盲點，我沒有注意到我用 POST，所以少查找了這一塊的資料 :)

**Comment by aszx87410** [comment]
不太確定 POST 是不是能被 cache：https://stackoverflow.com/questions/626057/is-it-possible-to-cache-post-methods-in-http

**Reply by Rubyrubylai** [reply]
我後來也有找到這一篇以及其他文章，看來若是要使用 POST 來做 cache 的設定會不太一樣，我會再研究一下，謝謝回覆 :)
https://tech.ebayinc.com/engineering/caching-http-post-requests-and-responses/
